huntsman/utils/pyro/serializers.py

```diff
@@ -56,8 +56,23 @@ def dict_to_astropy(class_name, d):
     return ayaml.load(d["yaml_dump"])
 
 
+def value_error_to_dict(obj):
+    """Serialiser function for ValueError."""
+    return {"__class__": "ValueError",
+            "args": [str(arg) for arg in obj.args]}
+
+
+def dict_to_value_error(class_name, d):
+    """De-serialiser function for ValueError."""
+    return ValueError(*d["args"])
+
+
 SerializerBase.register_class_to_dict(u.Quantity, astropy_to_dict)
 SerializerBase.register_dict_to_class("astropy_yaml", dict_to_astropy)
 
 SerializerBase.register_class_to_dict(error.PanError, panerror_to_dict)
 SerializerBase.register_dict_to_class("PanError", dict_to_panerror)
+
+# These two are only here as a temporary workaround for some typos in POCS
+SerializerBase.register_class_to_dict(ValueError, value_error_to_dict)
+SerializerBase.register_dict_to_class("ValueError", dict_to_value_error)
```

Comment:
In POCS there are a few instance of 

`raise ValueError("Blah, blah, blah, {}!", obj)`

where it should be

`raise ValueError("Blah, blah, blah, {}!".format(obj))`

or

`raise ValueError(f"Blah, blah, blah, {obj}!")`, e.g. 

https://github.com/panoptes/POCS/blob/22532dab81745aa5b805d809b4c494bd5495bc39/pocs/focuser/focuser.py#L242 and https://github.com/panoptes/POCS/blob/22532dab81745aa5b805d809b4c494bd5495bc39/pocs/focuser/focuser.py#L249 

These don't cause problems with local cameras (beyond funny looking exceptions) but it was causing test failures with the remote cameras because Pyro was trying to serialise `ValueErrors` that contained, in their `args` list, a reference to an object that it couldn't serialise. It took me a while to track down the root cause of the opaque 'circular reference' exceptions that Pyro was throwing as a result! My workaround in order to get this PR working was to add a custom serialiser/deserialiser for `ValueError` that simply casts all of the `args` to `str`, thereby making sure that `Pyro` can handle them. If we fix this in POCS then I can take this out again.

Reply:
https://github.com/panoptes/POCS/issues/966